Networking/Networking/Model/Product/ProductReviewStatus.swift

```diff
@@ -0,0 +1,81 @@
+/// Represents a ProductReviewStatus Entity.
+///
+public enum ProductReviewStatus: Decodable, Hashable {
+    case approved
+    case hold
+    case spam
+    case unspam
+    case trash
+    case untrash
+}
+
+
+/// RawRepresentable Conformance
+///
+extension ProductReviewStatus: RawRepresentable {
```

Comment:
just wondering, is `RawRepresentable` the pattern we follow instead of `enum ProductReviewStatus: String` with `enum Keys` as its raw value? like:

```swift
public enum ProductReviewStatus: String, Decodable {
    case approved = "approved"
    case hold = "hold"
    ...
}
```

Reply:
That's a good question. I lifted this from other model objects and I am not sure it is completely necessary in this case, as the keys and their raw values match. Maybe it's good to have that layer of indirection? I am not sure, to be completely honest. What do you think?

Reply:
I personally like less code hah, but a layer of indirection is good too. When I specify the raw value as the server value, I'd have to make a note in the comment so that the raw value doesn't get updated accidentally. The layer seems more clear that the value is from the server/other sources

Reply:
Yeah, I sure your point, less code (or more compact code, or less verbose code) is usually better. I don't know, to me, this is one of those cases where we could go either way.